models/pathology_nuclei_segmentation_classification/configs/inference.json

```diff
@@ -0,0 +1,150 @@
+{
+    "imports": [
+        "$import glob",
+        "$import os"
+    ],
+    "bundle_root": ".",
+    "output_dir": "$os.path.join(@bundle_root, 'eval')",
+    "dataset_dir": "/workspace/Data/CoNSeP/Test/Images",
+    "num_cpus": 2,
+    "batch_size": 1,
+    "sw_batch_size": 16,
+    "hovernet_mode": "original",
+    "patch_size": 270,
+    "out_size": 80,
+    "device": "$torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')",
+    "network_def": {
+        "_target_": "HoVerNet",
+        "mode": "@hovernet_mode",
+        "in_channels": 3,
+        "out_classes": 5
+    },
+    "network": "$@network_def.to(@device)",
+    "preprocessing": {
+        "_target_": "Compose",
+        "transforms": [
+            {
+                "_target_": "LoadImaged",
+                "keys": "image"
+            },
+            {
+                "_target_": "EnsureChannelFirstd",
+                "keys": "image"
+            },
+            {
+                "_target_": "CastToTyped",
+                "keys": "image",
+                "dtype": "float32"
+            },
+            {
+                "_target_": "ScaleIntensityRanged",
+                "keys": "image",
+                "a_min": 0.0,
+                "a_max": 255.0,
+                "b_min": 0.0,
+                "b_max": 1.0,
+                "clip": true
+            }
+        ]
+    },
+    "data_list": "$[{'image': image} for image in glob.glob(os.path.join(@dataset_dir, '*.png'))]",
+    "dataset": {
+        "_target_": "Dataset",
+        "data": "@data_list",
+        "transform": "@preprocessing"
+    },
+    "dataloader": {
+        "_target_": "DataLoader",
+        "dataset": "@dataset",
+        "batch_size": "@batch_size",
+        "shuffle": false,
+        "num_workers": "@num_cpus",
+        "pin_memory": true
+    },
+    "inferer": {
+        "_target_": "SlidingWindowHoVerNetInferer",
+        "roi_size": "@patch_size",
+        "sw_batch_size": "@sw_batch_size",
+        "overlap": "$1.0 - float(@out_size) / float(@patch_size)",
+        "padding_mode": "constant",
+        "cval": 0,
+        "sw_device": "@device",
```

Comment:
remove sw_device and device fields and let it be None.. so that during inferer input based device gets selected..
similar strategy is used in other bundles like speen

Reply:
In all the bundles if gpu available `@device` will b set to 'cuda:0' and the model will reside there, so why the input sliding window data (`sw_device`) should reside somewhere else?

Reply:
if device is None.. doesn't mean it's happening on CPU..
https://github.com/Project-MONAI/MONAI/blob/dev/monai/inferers/utils.py#L102-L108
https://github.com/Project-MONAI/MONAI/blob/dev/monai/inferers/utils.py#L129-L132

if input is already loaded into cuda.. things will happen accordingly
same for sw_device

Reply:
if you have observed when input in cuda but things happening on cpu.. then it's a bug.. u might want to create an issue the same on monai github

Reply:
so i would rather keep default based on the input rather force the device type.. even for deploy they only consider the transforms/inferers at the end.. and currently bundle api doesn't support loading particular section with overrides (to override the GPU selected by user)

I am referring `get_parsed_content(k, instantiate=True)` from monai bundle apis.. it doesnt support passing any overrides for the attributes.

Reply:
I look a bit more into it and it seems that it doesn't matter to set `sw_device` to `None` or `@device` since we are setting `device` argument in `SupervisedEvaluator` to `@device`, which sends the input to the same `@device`.

Reply:
I removed it anyways.